### DISCUSSION TITLE
Magick.NET Behavior In Azure App Service

### DISCUSSION THREAD

**Discussion by zdickinson** [dlemstra/Magick.NET#1273] [Help]
Hello. Our team has been working to remove System.Drawing functionality out of our app because things like "MetaFile" and "Bitmap" aren't able to be used in an Azure app service. Using Magick.NET-Q8-AnyCPU version 12.1.0 in our NET 6 web api we've attempted to get Wmf information and subsequent colors using the following.

```
var image = new MagickImage(imageStream, MagickFormat.Wmf);

var histogram = image.Histogram();
```

Running this code on a Windows machine provides the correct list of colors. When ran on the Azure App Service the only color returned is white, or 255-255-255, every time.

No errors seemed to be returned, and I can confirm the image stream I pass into MagickImage is the correct color as I save it out to a cloud bucket before performing the displayed code.

Is there anything I might be missing here? I can't even attach the test WMF because it's not a supported file type, which is pretty ironic.

**Comment by dlemstra** [comment]
You can attach the file here if you zip it. For the `Wmf` format GDI support is required and I don't know if that is available in an Azure App Service. I would expect that you get an exception when reading the image thought. My advise would be to save the `imageStream` somewhere and download it and make sure this file is correct.
